src/org/pentaho/reporting/platform/plugin/async/PentahoAsyncExecutor.java

```diff
@@ -108,16 +103,28 @@ private String getSessionId() {
     tasks.remove( key );
   }
 
+  @Override public void requestPage( final UUID id, final IPentahoSession session, final int page ) {
+    validateParams( id, session );
+    final IAsyncReportExecution<InputStream> runningTask = tasks.get( new CompositeKey( session, id ) );
+    if ( runningTask != null ) {
+      runningTask.requestPage( page );
+    }
+  }
+
   @Override public IAsyncReportState getReportState( final UUID id, final IPentahoSession session ) {
+    validateParams( id, session );
+    // link to running task
+    final IAsyncReportExecution<InputStream> runningTask = tasks.get( new CompositeKey( session, id ) );
+    return runningTask == null ? null : runningTask.getState();
+  }
+
+  private void validateParams( final UUID id, final IPentahoSession session ) {
     if ( id == null ) {
```

Comment:
You can safe some lines by using ArgumentNullException.validate(..) which also gives you a nice standard message.